packages/rx_bloc_cli/mason_templates/bricks/lib_auth_matrix/__brick__/test/lib_auth_matrix/services/auth_matrix_service_test.dart

```diff
@@ -0,0 +1,215 @@
+        ]),
+      );
+
+      // Test the change address actions
+      await expectLater(
+        authMatrixService.initiateAuthMatrix(payload: addressPayload),
+        emitsInOrder([
+          _Stub.responseOtp('2'),
+          _Stub.responseComplete('2'),
+          emitsDone,
+        ]),
+      );
+
+      // Test the [AuthMatrixService.onResponse] stream to contain all responses from both actions
+      expect(responses, [
+        _Stub.responsePinBiometric('1'),
+        _Stub.responseOtp('1'),
+        _Stub.responseComplete('1'),
+        _Stub.responseOtp('2'),
+        _Stub.responseComplete('2'),
+      ]);
+    });
+
+    test('initiateAuthMatrix emits complete method', () async {
+      final payload = AuthMatrixUnlockPayload();
+
+      when(
+        mockAuthMatrixRepository.initiate(
+          action: anyNamed('action'),
+          request: anyNamed('request'),
+        ),
+      ).thenAnswer((_) async => _Stub.responseComplete('1'));
+
+      final responses = <AuthMatrixResponse>[];
+      authMatrixService.onResponse.listen(responses.add);
+
+      await expectLater(
+        authMatrixService.initiateAuthMatrix(payload: payload),
+        emitsInOrder([
+          _Stub.responseComplete('1'),
+          emitsDone,
+        ]),
+      );
+
+      expect(responses, [
+        _Stub.responseComplete('1'),
+      ]);
+    });
+
+    test('initiateAuthMatrix handles router service errors', () async {
+      final payload = AuthMatrixUnlockPayload();
+
+      when(mockRouterService.push<Result<AuthMatrixResponse>>(
+        const AuthMatrixPinBiometricsRoute('1'),
+        extra: _Stub.responsePinBiometric('1'),
+      )).thenAnswer((_) async => Result.error(Exception('Router error')));
+
+      when(
+        mockAuthMatrixRepository.initiate(
+          action: anyNamed('action'),
+          request: anyNamed('request'),
+        ),
+      ).thenAnswer((_) async => _Stub.responsePinBiometric('1'));
+
+      final responses = <AuthMatrixResponse>[];
+      authMatrixService.onResponse.listen(responses.add);
+
+      await expectLater(
+        authMatrixService.initiateAuthMatrix(payload: payload),
+        emitsInOrder([
+          _Stub.responsePinBiometric('1'),
+          emitsError(isA<Exception>()),
+        ]),
+      );
+
+      expect(responses, [_Stub.responsePinBiometric('1')]);
+    });
+  });
+
+  test('initiateAuthMatrix handles repository errors', () async {
+    final payload = AuthMatrixUnlockPayload();
+
+    final responses = <AuthMatrixResponse>[];
+    authMatrixService.onResponse.listen(responses.add);
+
+    when(
+      mockAuthMatrixRepository.initiate(
+        action: anyNamed('action'),
+        request: anyNamed('request'),
+      ),
+    ).thenAnswer((_) async => throw Exception('Repository error'));
+
+    await expectLater(
+      authMatrixService.initiateAuthMatrix(payload: payload),
+      emitsError(isA<Exception>()),
+    );
+
+    expect(responses, []);
+  });
+}
+
+class _Stub {
```

Comment:
Do you think it's a good idea to have a separate `/mocks/stubs.dart` file where we would keep all stubs?
That's the approach I went with when writing the latest tests, I think it keeps the test code cleaner but we do run the risk of eventually having a mega-file with too many variables

Reply:
I think that with the stub class we can achieve reusability. In this particular case there is no such need.